packages/core/src/components/time-picker/time-picker.tsx

```diff
@@ -403,7 +402,7 @@ export class TimePicker {
     if (newUnitIndex >= 0 && newUnitIndex < this.timePickerDescriptors.length) {
       const newUnit = this.timePickerDescriptors[newUnitIndex].unit;
 
-      const selectedValue = Number(this.formattedTime[newUnit]);
+      const selectedValue = Number(this.formattedTime![newUnit]);
```

Comment:
You cannot simple overwrite with `!` operator. Lets have a look into the function `getFormattedTime` which return undefined.